theano/gpuarray/opt.py

```diff
@@ -193,7 +193,8 @@ def local_opt(node):
                 context_name = None
                 # We replace if any input is a host_from_gpu
                 for i in node.inputs:
-                    if i.owner and i.owner.op == host_from_gpu:
+                    if (i.owner and i.owner.op == host_from_gpu and
+                            move_to_gpu(i)):
```

Comment:
I'm good with that. I'm thinking that maybe around line 233, we should pre-move the inputs to the GPU while calling move_to_gpu? This way, maybe we could have more ops just return the op instance and let the op_lifter decide which inputs to move? Otherwise currently we need in the make_node of all gpu ops to move/don't move to the gpu explicitly.

Anyway, this is not for this PR. If you have an opinion, tell it. Probably not for this release.